### PR TITLE
Cleans up SAMLComplianceException factory methods

### PR DESCRIPTION
**Pull request by coyotesqrl**
Also adds an optional Node to be pretty-printed in the case of an exception being thrown.

### CODE REVIEW

File: library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt [hunk 1]
```diff
@@ -13,41 +13,52 @@
...
+
 class SAMLComplianceException : Exception {
 
     private constructor(message: String) : super(message)
     private constructor(message: String, cause: Throwable) : super(message, cause)
 
     companion object {
-        fun create(vararg codes: SAMLSpecRefMessage, message: String, cause: Throwable? = null):
+        fun create(vararg codes: SAMLSpecRefMessage,
+                   message: String,
```
Comment (bakejeyner) [inline library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt]: ❓ message is now required in `create`?

File: library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt [hunk 2]
```diff
@@ -13,41 +13,52 @@
...
         }
 
-        fun createWithPropertyReqMessage(section: String, property: String, parent: String): SAMLComplianceException {
-            return SAMLComplianceException("$section: $property is required in $parent.")
-        }
-
-        fun createWithPropertyInvalidMessage(code: SAMLSpecRefMessage, property: String, propertyValue: String?):
-                SAMLComplianceException {
-            return SAMLComplianceException("The $property value of $propertyValue is invalid.\n\n${readCode(code)}")
+        fun createWithPropertyReqMessage(section: String,
```
Comment (bakejeyner) [inline library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt]: ❓ Should we specify that `property` in this context means a property in an xml, and not some other property like in an object?

Maybe something like:
```
createWithXmlPropertyReqMessage
```

File: library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt [hunk 3]
```diff
@@ -13,41 +13,52 @@
...
-                                              property: String,
-                                              propertyValue: String?,
-                                              otherValue: String?): SAMLComplianceException {
-            return SAMLComplianceException("The $property value of $propertyValue is not equal to $otherValue." +
-                    "\n\n${readCode(code)}")
+        @Suppress("LongParameterList")
+        fun createWithPropertyMessage(code: SAMLSpecRefMessage,
+                                      property: String,
+                                      propertyValue: String?,
+                                      otherValue: String? = null,
```
Comment (blen-desta) [inline library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt]: ✏️ Can we rename `propertyValue` and `otherValue` it's not intuitive. Something like `actual` and `expected`.

File: library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt (hunk 1 above)
Comment (coyotesqrl) [reply library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt]: It always was. The only changes here are to add the `node` and to reformat the method to put each param on a line of its own.

File: library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt (hunk 2 above)
Comment (coyotesqrl) [reply library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt]: I can make that change.

**Pull request merged**